### DISCUSSION TITLE
[General] Missing a block when listening event locally

### DISCUSSION THREAD

**Discussion by enochSatis** [zkSync-Community-Hub/zksync-developers#181] [General]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

I have deployed a contract on Mainnet, and would like to listen to the contract's events to determine different interaction transactions. 

### What is the specific issue or error you're encountering?

When pinging zkSync's sequencer node (`https://mainnet.era.zksync.io/`), the L2 block `16634904` is missing in my local scanner. Particularly, the event of this transaction (`https://explorer.zksync.io/tx/0x062687f5b96d46188ee64093dd10952d1b4ff24ce1728e2251cb4b8cd1218e1a`) was missed. Was there any update/special behaviour during that period of time?

### Can you share the error messages or logs you're receiving, if any?

The event of `0x062687f5b96d46188ee64093dd10952d1b4ff24ce1728e2251cb4b8cd1218e1a` was not successfully caught. 

### Have you made any recent changes to the contract before encountering this issue?

No.

### Are there any external libraries or contracts that your contract interacts with?

No. 

### Can you provide the relevant portions of your contract code where the issue is occurring?

The target contract which get called (`0x5f3D22B90e98aAaA4267234D84485135a865915e`) triggers another contract which emits events. 

### Have you tried to isolate the problem, and if so, what were the results?

NA

### What steps have you already taken to try to resolve the issue?

NA

**Comment by FeichiDe** [comment]
cool！

**Comment by Romsters** [comment]
@enochSatis What API/utils did you use to listen for events?

**Reply by enochSatis** [reply]
I used the `eth_getLogs` method for RPC API call. Basically just listened to an address for all related events. Seems the event log will recover if I call the API later (I called the API several hours after the transaction happened and it recovered the missing log in that block), but when pinging the API in real time (every 2 secs), that particular event was missed out. 

Here is my Python script:
```
import requests

# Get logs through RPC node API call
def get_logs(address_list, from_block, to_block):
    header = {"Content-Type": "application/json"}
    post_obj = {
        "id": 1873156,
        "jsonrpc": "2.0",
        "method": "eth_getLogs",
        "params": [{
            "address": address_list,
            "fromBlock": hex(from_block),
            "toBlock": hex(to_block),
            "topics": []
        }]
    }
    res = requests.post(config.rpc_url, json=post_obj, headers=header, timeout=30)
    #logging.info(res.json())
    logs = res.json()['result']
    #logging.info(f"Obtained logs: {logs}")
    #logging.info("")
    return logs
```

The address I listened to is `0x61814F84cd631C617Bb925d0eD84bad55E84984f`. 

One of my colleagues also listened to this address using the same RPC API call method but with RUST. He had the same problem and the event in that block was missed.

**Reply by Romsters** [reply]
@enochSatis Could it be possible that you called the `eth_getLogs` API with a block range that included `16634904` block before it was actually sealed? Can you please share the code of how do you get `from_block` and  `to_block` and how the function is called every 2 seconds?
